web/Web/web/interfaces/DocumentSnippetEngine/functions.py

```diff
@@ -62,6 +63,38 @@ def get_documents(doc_ids, query=None, top_terms=None, orig_para_id=None):
     return result
 
 
+
+def get_document_or_none(doc_id):
```

Comment:
Is this function really needed? It seems like it has a lot of duplicate code from `get_documents()`

Reply:
I wrote it so if the document didn't exist (returns a none) we wouldn't send it to CAL. We could use `get_documents()` but I didn't want search the content for the words "404 not found". It felt a bit hacky. Another way we could do this is to add a function that checks if a document exists and just returns true or false.

Reply:
Hmm can we modify `get_documents` directly? Maybe update the title/content when there's a 404 and add another field `exists` in the `document` dict.  I'm hesitant to adding more functions in this file because anyone who uses the system would need to do more work to change it for whatever collection they want to use [(e.g. this is how I'm doing it for the `commoncrowl` collection)](https://github.com/UWaterlooIR/gathera/blob/cc-prod/web/Web/web/interfaces/DocumentSnippetEngine/functions.py#L62).

Reply:
I added an ok field to the docs and removed the extra function.